openshift/AGENTS.md

```diff
@@ -0,0 +1,727 @@
+# AGENTS.md - Guidance for AI Tools
+
+This document provides essential information for AI coding assistants working with the OpenShift operator-controller repository.
+
+---
+
+## Critical Context: This is a Fork
+
+This repository (`openshift/operator-framework-operator-controller`) is OpenShift's fork of upstream operator-controller (`operator-framework/operator-controller`). The maintainers work to minimize the diff between upstream and this fork. AI tools MUST understand the special constraints and workflows that govern changes to this codebase.
+
+**Upstream Repository:** https://github.com/operator-framework/operator-controller
+
+**For general operator-controller information**, refer to the upstream AGENTS.md file at the root of this repository, which is synced from upstream.
+
+---
+
+## Why This Matters for AI Tools
+
+Unlike most repositories where you can freely suggest changes anywhere in the codebase, this fork:
+
+- **Gets rebased regularly** against upstream operator-controller (with each upstream release)
+- **Must maintain minimal divergence** from upstream
+- **Has strict commit message conventions** that affect how changes survive rebases
+- **Requires changes to be categorized** as either upstream cherry-picks or downstream carries
+
+**Before suggesting any code changes, AI tools MUST understand the carry patch process documented below.**
+
+---
+
+## Understanding the Rebase Process
```

Comment:
Worth including that these processes are crystallized in [openshift/operator-framework-tooling](https://github.com/openshift/operator-framework-tooling)?